src/Controls/tests/DeviceTests/Elements/Window/WindowTests.Windows.cs

```diff
@@ -3,17 +3,64 @@
 using System.Threading.Tasks;
 using Microsoft.Maui.Controls;
 using Microsoft.Maui.DeviceTests.Stubs;
+using Microsoft.Maui.Graphics;
 using Microsoft.Maui.Graphics.Win2D;
 using Microsoft.Maui.Handlers;
 using Microsoft.Maui.Hosting;
 using Microsoft.Maui.Platform;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Microsoft.UI.Xaml.Automation.Provider;
 using Xunit;
 using WPanel = Microsoft.UI.Xaml.Controls.Panel;
 
 namespace Microsoft.Maui.DeviceTests
 {
 	public partial class WindowTests : ControlsHandlerTestBase
 	{
+		[Fact(DisplayName = "Swapping MainPage no Crash")]
+		public async Task SwappingMainPageNoCrash()
```

Comment:
I'm not sure if this test accurately tests the issue. 

I commented out all the changes in this PR and the test still passes.  One thing to note here is that the original issue was more about pushing the same page on the stack more then it was about swapping out the root page. 

So, the original issue is actually two different issues. 

1) is swapping out the rootpage https://github.com/drasticactions/MauiRepros/tree/main/MauiReusePageBug


2) is pushing a page, popping that page, pushing that page a second time.  https://github.com/Keflon/MauiNavigationBugRepro